src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1438,6 +1443,33 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
             return false;
         }
 
+        private static ArrowExpressionClauseSyntax GetExpressionBodyFromMember(MemberDeclarationSyntax containingMember)
+        {
+            if (containingMember != null)
+            {
+                switch (containingMember.Kind())
+                {
+                    case SyntaxKind.PropertyDeclaration:
+                        return ((PropertyDeclarationSyntax)containingMember).ExpressionBody;
+                    case SyntaxKind.MethodDeclaration:
+                        return ((MethodDeclarationSyntax)containingMember).ExpressionBody;
+                    case SyntaxKind.IndexerDeclaration:
+                        return ((IndexerDeclarationSyntax)containingMember).ExpressionBody;
+                    default:
+                        return null;
+                }
+            }
+
+            return null;
+        }
+
+        private static TextSpan ArrowExpressionClauseExpressionAdjustedSpan(ArrowExpressionClauseSyntax arrowExpressionClause)
```

Comment:
I don't like this name - adjusted for what?  This sounds like `InTrailingTrivia()` which we probably already have somewhere.